File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/iterators/RandomPhotoIterator.java
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.iterators;

/**
 * Created on 17/10/2018.
 */
public class RandomPhotoIterator extends AbstractPhotoIterator {

    @Override
    protected int getNextIndex() {
        return (int) (Math.floor(getTotalCount() * Math.random()));
    }
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/data_impl/photodata/gateway/greendao/iterators/RandomPhotoIterator.kt
package nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.iterators

import nl.acidcats.tumblrlikes.data_impl.photodata.gateway.greendao.filters.FilterOption

/**
 * Created on 25/10/2018.
 */
class RandomPhotoIterator constructor(filterOption: FilterOption) : AbstractPhotoIterator(filterOption) {
    override val nextIndex: Int
        get() = Math.floor(totalCount * Math.random()).toInt()
}